paimon-trino-common/src/main/java/org/apache/paimon/trino/TrinoFilterConverter.java

```diff
@@ -103,10 +107,20 @@ public Optional<Predicate> convert(
             TrinoColumnHandle columnHandle = entry.getKey();
             Domain domain = entry.getValue();
             String field = columnHandle.getColumnName();
+            Optional<Integer> nestedColumn = getNestedColumn(field);
+            if (nestedColumn.isPresent()) {
+                int position = nestedColumn.get();
+                field = field.substring(0, position);
+            }
             int index = fieldNames.indexOf(field);
             if (index != -1) {
                 try {
-                    conjuncts.add(toPredicate(index, columnHandle.getTrinoType(), domain));
+                    conjuncts.add(
+                            toPredicate(
+                                    index,
+                                    columnHandle.getColumnName(),
+                                    columnHandle.getTrinoType(),
+                                    domain));
```

Comment:
We should pass the nested field to the predicate. For example, new LeafPredicate(columnName="map_data[key1]")